onnxruntime/core/framework/session_state.cc

```diff
@@ -308,4 +456,38 @@ const NodeIndexInfo& SessionState::GetNodeIndexInfo() const {
   ORT_ENFORCE(node_index_info_, "SetGraphAndCreateKernels must be called prior to GetExecutionInfo.");
   return *node_index_info_;
 }
+
+void SessionState::UpdateToBeExecutedNodes(const std::vector<int>& fetch_mlvalue_idxs) {
+  if (to_be_executed_nodes_.find(fetch_mlvalue_idxs) != to_be_executed_nodes_.end())
+    return;
+
+  const Graph* graph = GetGraphViewer()->GetGraph();
+
+  // Get the nodes generating the fetches.
+  std::vector<const Node*> nodes;
+  nodes.reserve(fetch_mlvalue_idxs.size());
+  std::unordered_set<NodeIndex> reachable_nodes;
+
+  for (auto idx : fetch_mlvalue_idxs) {
+    std::string node_arg_name;
+    if (!this->GetOrtValueNameIdxMap().GetName(idx, node_arg_name).IsOK()) {
+      to_be_executed_nodes_.insert(std::make_pair(fetch_mlvalue_idxs, reachable_nodes));
```

Comment:
when would this fail? I would have expected any valid set of fetches to have any entry in the map.